### PR TITLE
feat: added custom time parsing (keptn/keptn#4788)

### PR DESCRIPTION
**Pull request by RealAnna**
## This PR

-  change time parsing to a custom one to avoid parsing errors
-  In the order we try :
      - ISO8601 :  000Z format
      -  then RFC3339 : "2019-10-12T07:20:50.52Z" 
      - finally a fallback userfriendly format : "2006-01-02T15:04:05"
     

### Related Issues

https://github.com/keptn/keptn/issues/4788

## Notes
RFC 3339 is following the ISO 8601 DateTime format. The only difference is RFC allows us to replace “T” with “space”

### CODE REVIEW

File: pkg/common/timeutils/timeutils.go [hunk 1]
```diff
@@ -55,13 +60,16 @@ func (params *GetStartEndTimeParams) Validate() error {
...
+
+	/*if params.TimeFormat == "" {
 		timeFormat = KeptnTimeFormatISO8601
+
 	} else {
 		timeFormat = params.TimeFormat
-	}
-	var err error
+	}*/
+
```
Comment (thisthat) [inline pkg/common/timeutils/timeutils.go:72]: Please, remove this unused code

File: pkg/common/timeutils/timeutils.go [hunk 2]
```diff
@@ -81,23 +89,23 @@ func GetStartEndTime(params GetStartEndTimeParams) (*time.Time, *time.Time, erro
 	// initialize default values for end and start time
 	end := time.Now().UTC()
 	start := time.Now().UTC().Add(-timeframeDuration)
-
+	var temp *time.Time
```
Comment (thisthat) [inline pkg/common/timeutils/timeutils.go:92]: why this temp var?

File: pkg/common/timeutils/timeutils.go [hunk 3]
```diff
@@ -18,13 +17,19 @@ func GetKeptnTimeStamp(timestamp time.Time) string {
 }
 
 // ParseTimestamp tries to parse the given timestamp using the ISO8601 format (e.g. '2006-01-02T15:04:05.000Z')
-// if this is not possible, the fallback format '2006-01-02T15:04:05.000000000Z' will be used. If this fails as well, an error is returned
+// if this is not possible, the fallback format RFC3339  will be used, then '2006-01-02T15:04:05'
+//If this fails as well, an error is returned
```
Comment (thisthat) [inline pkg/common/timeutils/timeutils.go:21]:
```suggestion
// if this is not possible, the fallback format RFC3339  will be used, then '2006-01-02T15:04:05'.
// If this fails as well, an error is returned
```

File: pkg/common/timeutils/timeutils.go [hunk 4]
```diff
@@ -78,41 +79,44 @@ func GetStartEndTime(params GetStartEndTimeParams) (*time.Time, *time.Time, erro
...
-	end := time.Now().UTC()
-	start := time.Now().UTC().Add(-timeframeDuration)
+	// calculate possible start and end time
+	now := time.Now().UTC()
+	calcStart := now.Add(-timeframeDuration)
+
+	// initialize default values for end and start time pointer
+	var end, start *time.Time
+	end = &now
+	start = &calcStart
```
Comment (thisthat) [inline pkg/common/timeutils/timeutils.go:89]: I think the previous code it is easier to read, but I would have it with some of your changes
```suggestion
	end := time.Now().UTC()
	start := end.Add(-timeframeDuration)
```

File: pkg/common/timeutils/timeutils.go [hunk 5]
```diff
@@ -55,13 +60,9 @@ func (params *GetStartEndTimeParams) Validate() error {
 
 // GetStartEndTime parses the provided start date, end date and/or timeframe
 func GetStartEndTime(params GetStartEndTimeParams) (*time.Time, *time.Time, error) {
-	var timeFormat string
-	if params.TimeFormat == "" {
-		timeFormat = KeptnTimeFormatISO8601
-	} else {
-		timeFormat = params.TimeFormat
-	}
+	//var timeFormat string
```
Comment (warber) [inline pkg/common/timeutils/timeutils.go:63]: pls remove this line

**Review by warber — CHANGES_REQUESTED** [review]

**Review by warber — APPROVED** [review]

**Pull request merged**